src/app/components/Document/index.jsx

```diff
@@ -3,16 +3,49 @@ import ResourceHints from './ResourceHints';
 import '../../lib/globalStyles';
 import { C_POSTBOX } from '../../lib/constants/styles';
 
+const isAmpPayload = htmlAttributes =>
+  htmlAttributes.toString().includes('amp');
+
+/* eslint-disable react/no-danger */
+/* disabling the rule that bans the use of dangerouslySetInnerHTML until a more appropriate implementation can be implemented */
+const injectClientSideApp = serialisedData => {
+  if (serialisedData) {
+    return (
+      <script
+        dangerouslySetInnerHTML={{
+          __html: `window.SIMORGH_DATA=${serialisedData}`,
+        }}
+      />
+    );
+  }
+
+  return null;
+};
+
 /* eslint-disable react/prop-types */
 const Document = ({ assets, app, data, styleTags, helmet }) => {
   const htmlAttrs = helmet.htmlAttributes.toComponent();
   const meta = helmet.meta.toComponent();
   const title = helmet.title.toComponent();
   const links = helmet.link.toComponent();
-  const serialisedData = JSON.stringify(data);
-  const scripts = assets.map(asset => (
-    <script key={asset} type="text/javascript" src={asset} defer />
-  ));
+  let scripts = [];
+  let ampScript;
+  let serialisedData;
+
+  if (isAmpPayload(helmet.htmlAttributes)) {
+    ampScript = (
+      <script key="amp" async src="https://cdn.ampproject.org/v0.js" />
+    );
+  }
+
+  if (!isAmpPayload(helmet.htmlAttributes)) {
```

Comment:
Rather than do this logic here, [data](https://github.com/bbc/simorgh/blob/latest/src/server/index.jsx#L65) returns an [`amp` boolean](https://github.com/bbc/simorgh/blob/latest/src/app/routes/getInitialData/index.js#L45), which would be cleaner to use.

We could extract all of the script creation to that level, including the creation of of the data script, async AMP script, and deferred bundle scripts. It can all go in a single `script` array, which can be pased to the Document for it to [`use`](https://github.com/bbc/simorgh/pull/718/files#diff-00a291acea0cc37f44c9f1daaa1fe90cL40).